x-pack/plugins/alerting/server/application/rule/methods/bulk_enable/bulk_enable_rules.ts

```diff
@@ -108,13 +108,7 @@ export const bulkEnableRules = async <Params extends RuleParams>(
     filter: kueryNodeFilterWithAuth,
   });
 
-  const [taskIdsToEnable] = accListSpecificForBulkOperation;
```

Comment:
I don't remember why we wanted to change this to update the rule first - according to the [table I put in the previous PR](https://github.com/elastic/kibana/pull/189041#issuecomment-2263944647), we should be updating the task first.

I'm wondering if we changed this back to the original order, so it only has the filter-y changes for enabled like the `bulk_disable_rules` module has - would the security rules start passing again?

Reply:
I think the code structure might be confusing here, because we are updating the task first as decided. Either by `scheduling new task at line 319` or by `enabling existing task at line 325`. And then we `update the rule at line 331`.

> I'm wondering if we changed this back to the original order, so it only has the filter-y changes for enabled like the bulk_disable_rules module has - would the security rules start passing again?

yes keeping only filter changes as per [first commit](https://github.com/elastic/kibana/pull/191492/commits/eaf2ec0fe4298a4411a02a97d2b254a1ad291221#diff-c6f68d894425c178e73474f7ea2b9b9480a3c143ddb4f6112d7e2e6752ca35d3) passes security tests 😄

Reply:
It is because in the original code `bulkEnableRulesWithOCC` returned taskIdsToEnable in `accListSpecificForBulkOperation`. But now since we do task enable before rule, `tryToEnableTasks` is called within `bulkEnableRulesWithOCC` and returns `taskIdsFailedToBeEnabled `.

Reply:
Ah ... I was confused looking at the code ...

Yeah, there must be something subtle going on there - I think we can leave the order alone for now - it will result in possibly having a zombie rule that looks enabled, but the task isn't, if there's a failure between the rule and task updates.  We'll fix that in the resilience issue/PR.  We didn't open an issue for that yet, did we?

So I'm happy just taking the first commit, if that was just the filtering bits ...